File: app/models/bottle.js
module.exports = function(db) {
    try {
        return db.model("Bottle");
    } catch(e) {}

    var ObjectId = db.schema.Types.ObjectId;

    var bottleSchema = new db.schema({

    	created: {type: Date, "default": Date.now},

        modified: Date,

        wine: {type: ObjectId, ref: "Wine"},

    	store: {type: ObjectId, ref: "Store"},

    	buyAgain: {type: Boolean, "default": false},

    	doWeLike: {type: Boolean, "default": false},

        notes: String,

        price: Number,

        datePurchased: Date,

        dateDrank: Date,

        memoryCue: String
    });

    return db.model("Bottle", bottleSchema);
}

File: models/bottle.go
package models

import (
	"gopkg.in/mgo.v2"
	"gopkg.in/mgo.v2/bson"
	"time"
)

type Bottle struct {
	Id           bson.ObjectId `bson:"_id"`
	CreatedDate  time.Time
	ModifiedDate time.Time
	Wine         bson.ObjectId
	Store        bson.ObjectId
	Rating       int
	BuyAgain     bool
	//Is this the proper type
	Price         float64
	DatePurchased time.Time
	DateDrank     time.Time
	MemoryCue     string
	Year          int
	Notes         string
	OnSale        bool
}

// func (b *Bottle) Update(wine bson.ObjectId, store bson.ObjectId, notes string, memoryCue string, buyAgain bool, doWeLike bool, price float64, datePurchased time.Time, dateDrank time.Time, year int, db *mgo.Database) {
// 	//need to check if variables present?
// 	b.ModifiedDate = time.Now()
// 	b.Notes = notes
// 	b.MemoryCue = memoryCue
// 	b.Wine = wine
// 	b.Store = store
// 	b.BuyAgain = buyAgain
// 	b.DoWeLike = doWeLike
// 	b.Price = price
// 	b.DatePurchased = datePurchased
// 	b.DateDrank = dateDrank
// 	b.Year = year
// 	b.Save(db)
// }

func (b *Bottle) Save(db *mgo.Database) error {
	_, err := b.coll(db).UpsertId(b.Id, b)
	return err
}

func (b *Bottle) FindByID(id bson.ObjectId, db *mgo.Database) error {
	return b.coll(db).FindId(id).One(b)
}

func (*Bottle) coll(db *mgo.Database) *mgo.Collection {
	return db.C("bottle")
}
